# With the introduction of Solaris 10 all pthread code has been moved to libc.  
# Linking to libpthread is now unnecessary and in some cases can cause issues.
--- ./configure.ac.bak	2011-07-18 10:00:30.028456125 +0100
+++ ./configure.ac	2011-07-18 10:07:21.588685016 +0100
@@ -1842,12 +1842,12 @@
                          pthread_mutex_init (&m, pthread_mutexattr_default);],
 			have_threads=dce)
         fi
-	# Tru64Unix requires -pthread to find pthread.h. See #103020
-	CPPFLAGS="$CPPFLAGS -pthread"
 	if test "x$have_threads" = xno; then
-	AC_TRY_COMPILE([#include <pthread.h>],
-		       [pthread_mutex_t m = PTHREAD_MUTEX_INITIALIZER;],
-		       have_threads=posix)
+                # Tru64Unix requires -pthread to find pthread.h. See #103020
+                CPPFLAGS="$CPPFLAGS -pthread"
+                AC_TRY_COMPILE([#include <pthread.h>],
+                        [pthread_mutex_t m = PTHREAD_MUTEX_INITIALIZER;],
+                        have_threads=posix)
         fi
 	CPPFLAGS="$glib_save_CPPFLAGS"
 fi
@@ -1914,8 +1914,8 @@
     # One of them does for most gcc versions and some other platforms/compilers
     # too and could be considered as the canonical way to go. 
     case $host in
-      *-*-cygwin*|*-*-darwin*)
-         # skip cygwin and darwin -pthread or -pthreads test
+      *-*-cygwin*|*-*-darwin*|*-solaris2.11)
+         # skip cygwin, darwin and solaris 10 -pthread or -pthreads test
          ;;
       *-solaris*)
         # These compiler/linker flags work with both Sun Studio and gcc
